src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -668,22 +695,37 @@ fn env_args(
             });
     }
 
-    if !rustflags.is_empty() {
-        return Ok(rustflags);
+    if rustflags.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(rustflags))
     }
+}
+
+fn rustflags_from_host(
+    config: &Config,
+    name: &str,
+    host_triple: &str,
+) -> CargoResult<Option<Vec<String>>> {
+    let target_cfg = config.host_cfg_triple(host_triple)?;
+    let list = if name.eq_ignore_ascii_case("rustflags") {
+        &target_cfg.rustflags
+    } else {
+        // host.rustdocflags is not a thing, since it does not make sense
+        return Ok(None);
+    };
+    Ok(list.as_ref().map(|l| l.val.as_slice().to_vec()))
+}
 
+fn rustflags_from_build(config: &Config, name: &str) -> CargoResult<Option<Vec<String>>> {
     // Then the `build.rustflags` value.
     let build = config.build_config()?;
-    let list = if name == "rustflags" {
+    let list = if name.eq_ignore_ascii_case("rustflags") {
```

Comment:
Since `name` is controlled at compile time it seems like this should be able to use either `RUSTFLAGS` or `rustflags` and match what's expected here. Otherwise this seems like it's over-generalizing for something that isn't used here?

Reply:
Good call — I changed it to be an `enum` instead to avoid all this unnecessary string business.